### PR TITLE
[Paddle Inference]Enhance the shape check of trt_embedding_eltwise_layernorm_fuse_pass,…

### PR DESCRIPTION
**Pull request by Wangzheee**
### PR types
Others

### PR changes
Others

### Description
Pcard-71501
- Enhance the shape check: trt_embedding_eltwise_layernorm_fuse_pass, don't support broadcast
- Enhance the shape check: embedding_eltwise_layernorm_fuse_pass, don't support broadcast

image

- fix test for trt_embedding_eltwise_layernorm_fuse_pass, embedding_eltwise_layernorm_fuse_pass

### CODE REVIEW

File: test/ir/inference/test_emb_eltwise_layernorm_fuse_pass.py [hunk 1]
```diff
@@ -236,81 +235,282 @@ def sample_predictor_configs(self, program_config):
...
-                    "input_data3": [4, 512, 1],
-                },
-                {
-                    "input_data1": [2, 128, 1],
-                    "input_data2": [2, 128, 1],
-                    "input_data3": [2, 128, 1],
-                },
+
+
+class TestEmbeddingEltwiseLayerNormFusePassNoBroadcast(PassAutoScanTest):
```
Comment (Wangzheee) [inline test/ir/inference/test_emb_eltwise_layernorm_fuse_pass.py:214]: 测试elementwise_add需要广播的情况

File: test/ir/inference/test_emb_eltwise_layernorm_fuse_pass.py [hunk 2]
```diff
@@ -17,11 +17,9 @@
...
 import hypothesis.strategies as st
 import numpy as np
-from auto_scan_test import IgnoreReasons, PassAutoScanTest
+from auto_scan_test import PassAutoScanTest
 from program_config import OpConfig, ProgramConfig, TensorConfig
 
-import paddle.inference as paddle_infer
-
 
 class TestEmbeddingEltwiseLayerNormFusePass(PassAutoScanTest):
```
Comment (Wangzheee) [inline test/ir/inference/test_emb_eltwise_layernorm_fuse_pass.py:24]: 只测试原生GPU的pass融合

File: test/ir/inference/test_emb_eltwise_layernorm_fuse_pass.py [hunk 3]
```diff
@@ -236,81 +235,282 @@ def sample_predictor_configs(self, program_config):
...
-                    "input_data3": [2, 128],
-                },
+
+        def generate_weight2(attrs):
+            return np.random.uniform(1, 1.1, attrs[3]['weight_size'][1]).astype(
+                np.float32
             )
-        yield config, ['fused_embedding_eltwise_layernorm'], (1e-5, 1e-5)
 
-    def add_ignore_pass_case(self):
```
Comment (Wangzheee) [inline test/ir/inference/test_emb_eltwise_layernorm_fuse_pass.py:298]: 去除了之前所有的ignore_check_case（跳过精度检查），对计算结果精度也进行检查

File: test/ir/inference/test_trt_emb_eltwise_layernorm_fuse_pass.py [hunk 4]
```diff
@@ -0,0 +1,620 @@
...
+
+import hypothesis.strategies as st
+import numpy as np
+from auto_scan_test import PassAutoScanTest
+from program_config import OpConfig, ProgramConfig, TensorConfig
+
+import paddle.inference as paddle_infer
+
+
+class TestEmbeddingEltwiseLayerNormFusePass(PassAutoScanTest):
```
Comment (Wangzheee) [inline test/ir/inference/test_trt_emb_eltwise_layernorm_fuse_pass.py:26]: 只测试 trt pass

File: test/ir/inference/test_trt_emb_eltwise_layernorm_fuse_pass.py [hunk 5]
```diff
@@ -0,0 +1,620 @@
...
+        # this fuse need to fix, now there's no program can ran successfully
+        self.run_and_statis(
+            quant=False,
+            max_examples=50,
+            passes=["trt_embedding_eltwise_layernorm_fuse_pass"],
+            min_success_num=0,
+        )
+
+
+class TestEmbeddingEltwiseLayerNormFusePassNoBroadcast(PassAutoScanTest):
```
Comment (Wangzheee) [inline test/ir/inference/test_trt_emb_eltwise_layernorm_fuse_pass.py:260]: 测试elementwise_add需要广播的情况

**Review by qingqing01 — DISMISSED** [review]

**Review by XieYunshen — APPROVED** [review]

**Pull request merged**